src/blog/2024/08/comparing-dashboard-2-with-uibuilder.md

```diff
@@ -0,0 +1,233 @@
+
+![Screenshot of the UI-Builder GitHub commit chart](./images/ui-builder-commits.png)  
+_Screenshot of the UI-Builder GitHub commit chart._
+
+## How Extensive is the Collection of UI Elements?
+
+### Node-RED Dashboard 2.0
+
+Node-RED Dashboard 2.0 offers an extensive set of UI elements, including forms, dropdowns, tables, charts, and gauges that are super easy to use. These widgets are built with complex Vue.js components, but users are completely shielded from this complexity, allowing them to focus on ease of use. Additionally, if you want to create your own custom widget, you can do so with the ui-template node that accepts complete Vue components. By default, Dashboard 2.0 supports the Vuetify library, which provides an extensive set of UI components.
+
+## UI-Builder
+
+UI-Builder also offers a number of widgets, but these are not as user-friendly as those in Dashboard 2.0. Users must send a JSON config object, which can be complex for new users who lack knowledge of HTML/CSS. Additionally, handling widget data requires using UI-Builder's methods, which can further increase complexity. However, UI-Builder's strength lies in its flexibility, allowing any HTML element to be used as a component, and it also supports the W3C standard web components. Despite this, users need to perform a lot of additional tasks to get everything set up and will have the hassle of writing things.
+
+## How much dynamic insertion or update it support?
+
+### Node-RED Dashboard
+
+Node-RED Dashboard 2.0 supports dynamically updating the UI via the msg object. Each UI widget supports the `msg.ui_update` property with a JSON object that allows you to update the UI components dynamically. For example, you can update form fields based on user input, dynamically insert or update options in a dropdown, or change the appearance of the UI by sending css classes.
+
+### Ui-builder
+
+UI-Builder also supports dynamic UI updates. Similar to Node-RED Dashboard, you can use messages to control the state and content of UI elements.
+
+## Is creating visualizations easy in the Dashboard? ?
+
+### Node-RED Dashboard 2.0
+
+Visualization is a core use case for Node-RED, and Node-RED Dashboard 2.0 makes it easy with its built-in [chart](https://dashboard.flowfuse.com/nodes/widgets/ui-chart.html) and [gauge widgets](https://dashboard.flowfuse.com/nodes/widgets/ui-gauge.html). The chart widgets support various types, including line, bar, scatter, pie, and donut charts, while the gauge widgets offer options like tile, 3/4 gauge, and half gauge. This range of built-in options simplifies the creation of visualizations. Additionally, you can use Vuetify or other third-party libraries in the ui-template node if you need different types of charts.
```

Comment:
_Visualization is a core use case for Node-RED_

No it isn't... Node-RED doesn't do visualisation _without_ a third party node set like Dashboard.

Reply:
i mean monitoring, will update